reverse_replication/ReverseReplicationUserGuide.md

```diff
@@ -0,0 +1,187 @@
+mean(value.user_counter)]
+| every 1m
+```
+
+Metrics visible on Dataflow UI  can also be queried via REST,official document [here](https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.locations.jobs/getMetrics?apix_params=%7B%22projectId%22%3A%22span-cloud-testing%22%2C%22location%22%3A%22us-east1%22%2C%22jobId%22%3A%222023-06-06_05_20_27-10999367971891038895%22%7D).
+
+### Troubleshooting
+
+The Dataflow logs are an excellent way to understand if something is not working as expected.
+
+If you observe that the pipeline is not making expected progress, check the Dataflow logs for any errors.For Dataflow related errors, please refer [here](https://cloud.google.com/dataflow/docs/guides/troubleshooting-your-pipeline) for troubleshooting.
+
+Apart from that,following are some scenarios and how to handle them.
+
+- ***The watermark of the Spanner to Sink pipeline does not advance***
+
+    This happens when the job is hit with a huge backlog, that leads to infinite loop. The recovery steps are covered [here](README.md).
+
+- ***PubSub message count is not decreasing and the same data is being written back to source***
+
+    This happens when the time to write all messages to source database and send an ACK to PubSub exceeds the deadline. Ensure the ACK deadline for the subscriptions are high enough (10 minutes is the highest value, consider bumping it to that). If still facing this issue, consider moving the Dataflow job writing to source database geographically closer to the source database.
+
+- ***Some records are not reverse replicated***
+
+    Records of below nature are dropped from reverse replication. Check the Dataflow logs to see if they are dropped.
+    1. Records which are forward migrated. 
+    2. Records for which primary key cannot be determinted on the source database.This can happen when the source database table does not have a primary key, or the primary key value was not present in the change stream data, or the record was deleted on Cloud Spanner and the deleted record was removed from Cloud Spanner due to lapse of retention period by the time the record was to be reverse replicated.
+
+- ***There is higher load than the expected QPS on  spanner instance post cutover***
+
+     If the forward migration is still running post cutover, the incoming writes on Spanner that are reverse-replicated to the source get forward migrated again. This can cause the load on Spanner to be almost double the expected QPS, as each write will get reflected twice. Also, it could lead to transient inconsistencies in data under certain cases. To avoid this, stop/delete the forward migration pipeline post cutover. If the forward pipeline is required, add custom filtration rules and build a custom forward migration dataflow template.
+
+### Retry
+
+For both the Dataflow jobs, once an error is encountered for a given shard , then procesing is stopped for that shard to preserve ordering.The way to recover is to rerun the job.
```

Comment:
[super nit]: 
1. Extra space before `,`.
2. Would it be more direct to say `To recover, rerun the job.`
3. Should we add a note that the jobs are idempotent to emphasise that it's safe to rerun the jobs?

Reply:
Done